src/org/daisy/dotify/formatter/impl/page/PageImpl.java

```diff
@@ -223,6 +226,17 @@ void addIdentifiers(List<String> ids) {
 		return identifiers;
 	}
 	
+	private void startsContentIdentifiers() {
+		contentIdentifiersBegin = getIdentifiers().size();
+	}
+	
+	/**
+	 * Get identifiers for this page excluding identifiers before text content
+	 */
+	public List<String> getContentIdentifiers() {
+		return getIdentifiers().subList(contentIdentifiersBegin, getIdentifiers().size());
```

Comment:
Instead of returning the content identifiers, you may consider returning the _other_ identifiers
`return getIdentifiers().subList(0, contentIdentifiersBegin);`

In this way you won't need a negation in `VolumeProvider.java`. Instead of `boolean atStartOfPage = !contentIdentifiers.contains(id);` you can use something like `boolean atStartOfPage = startIdentifiers.contains(id)`; (or `initialIdentifiers` or so)